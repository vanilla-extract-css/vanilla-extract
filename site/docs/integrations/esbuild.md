---
title: esbuild
parent: integrations
---

# esbuild

A plugin for integrating vanilla-extract with [esbuild](https://esbuild.github.io/).

## Installation

```bash
npm install --save-dev @vanilla-extract/esbuild-plugin
```

## Setup

Add the plugin to your build script, along with any desired [plugin configuration](#configuration).

```js
// bundle.js

const {
  vanillaExtractPlugin
} = require('@vanilla-extract/esbuild-plugin');

require('esbuild')
  .build({
    entryPoints: ['app.ts'],
    bundle: true,
    plugins: [vanillaExtractPlugin()],
    outfile: 'out.js'
  })
  .catch(() => process.exit(1));
```

## Configuration

```js
// bundle.js

const {
  vanillaExtractPlugin
} = require('@vanilla-extract/esbuild-plugin');

require('esbuild').build({
  plugins: [
    vanillaExtractPlugin({
      // configuration
    })
  ]
});
```

The plugin accepts the following as optional configuration:

### processCss

As esbuild currently doesn't have a way to process the CSS generated by vanilla-extract, you can optionally use the `processCss` option.

For example, to run autoprefixer over the generated CSS.

```js
// bundle.js

const {
  vanillaExtractPlugin
} = require('@vanilla-extract/esbuild-plugin');
const postcss = require('postcss');
const autoprefixer = require('autoprefixer');

async function processCss(css) {
  const result = await postcss([autoprefixer]).process(
    css,
    {
      from: undefined /* suppress source map warning */
    }
  );

  return result.css;
}

require('esbuild')
  .build({
    entryPoints: ['app.ts'],
    bundle: true,
    plugins: [
      vanillaExtractPlugin({
        processCss
      })
    ],
    outfile: 'out.js'
  })
  .catch(() => process.exit(1));
```

### identifiers

Different formatting of identifiers (e.g. class names, keyframes, CSS Vars, etc) can be configured by selecting from the following options:

- `short` identifiers are a 7+ character hash. e.g. `hnw5tz3`
- `debug` identifiers contain human readable prefixes representing the owning filename and a potential rule level debug name. e.g. `myfile_mystyle_hnw5tz3`

Each integration will set a default value based on the configuration options passed to the bundler.

### esbuildOptions

esbuild is used internally to compile `.css.ts` files before evaluating them to extract styles. You can pass additional options here to customize that process.
Accepts a subset of esbuild build options (`plugins`, `external`, `define`, `loader` and `tsconfig`). See the [build API](https://esbuild.github.io/api/#build-api) documentation.
