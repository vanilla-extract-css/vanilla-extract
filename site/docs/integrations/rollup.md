---
title: Rollup
parent: integrations
---

# Rollup

A plugin for integrating vanilla-extract with [Rollup](https://rollupjs.org/).

This plugin is useful for library development but not suitable for application bundles.
Rollup has no built-in CSS bundling, so this plugin just outputs styles as individual CSS assets.

For applications we instead recommend to use Vite
(which itself uses Rollup under the hood but comes with its own CSS bundling).

## Installation

```bash
npm install --save-dev @vanilla-extract/rollup-plugin
```

## Setup

Add the plugin to your Rollup configuration, along with any desired [plugin configuration](#configuration).

```js
// rollup.config.js
import { vanillaExtractPlugin } from '@vanilla-extract/rollup-plugin';

export default {
  plugins: [vanillaExtractPlugin()]
};
```

This plugin works well with Rollup's `preserveModules`.

Rollup by default places assets in an "assets" directory.
You can configure [asset file names](https://rollupjs.org/guide/en/#outputassetfilenames)
if you care about CSS assets being placed right next to the corresponding JS files.

```js
// rollup.config.js
import { vanillaExtractPlugin } from '@vanilla-extract/rollup-plugin';

export default {
  plugins: [vanillaExtractPlugin()],
  output: {
    preserveModules: true,
    assetFileNames({ name }) {
      return name?.replace(/^src\//, '') ?? '';
    }
  }
};
```

## Configuration

```js
// rollup.config.js
import { vanillaExtractPlugin } from '@vanilla-extract/rollup-plugin';

export default {
  plugins: [
    vanillaExtractPlugin({
      // configuration
    })
  ]
};
```

The plugin accepts the following as optional configuration:

### identifiers

Different formatting of identifiers (e.g. class names, keyframes, CSS Vars, etc) can be configured by selecting from the following options:

- `short` identifiers are a 7+ character hash. e.g. `hnw5tz3`
- `debug` identifiers contain human readable prefixes representing the owning filename and a potential rule level debug name. e.g. `myfile_mystyle_hnw5tz3`

Each integration will set a default value based on the configuration options passed to the bundler.
