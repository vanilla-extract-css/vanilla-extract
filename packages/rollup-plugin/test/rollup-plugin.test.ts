import { rollup, OutputOptions } from 'rollup';
import esbuild from 'rollup-plugin-esbuild';
import json from '@rollup/plugin-json';
import path from 'path';

import { vanillaExtractPlugin } from '..';

async function buildAndMatchSnapshot(outputOptions: OutputOptions) {
  const bundle = await rollup({
    input: require.resolve('@fixtures/themed'),
    plugins: [
      vanillaExtractPlugin({
        cwd: path.dirname(require.resolve('@fixtures/themed/package.json')),
      }),
      esbuild(),
      json(),
    ],
    external: ['@vanilla-extract/dynamic'],
  });
  const { output } = await bundle.generate(outputOptions);
  output.sort((a, b) => a.fileName.localeCompare(b.fileName));
  expect(
    output.map((chunkOrAsset) => [
      chunkOrAsset.fileName,
      chunkOrAsset.type === 'asset' ? chunkOrAsset.source : chunkOrAsset.code,
    ]),
  ).toMatchSnapshot();
  if (bundle) {
    await bundle.close();
  }
}

describe('rollup-plugin', () => {
  it('should build without preserveModules', async () => {
    // Bundle all JS outputs together
    await buildAndMatchSnapshot({
      format: 'esm',
    });
  });

  it('should build with preserveModules', async () => {
    // Preserve JS modules
    await buildAndMatchSnapshot({
      format: 'esm',
      preserveModules: true,
    });
  });

  it('should build with preserveModules and assetFileNames', async () => {
    // Preserve JS modules and place assets next to JS files instead of assets directory
    await buildAndMatchSnapshot({
      format: 'esm',
      preserveModules: true,
      assetFileNames({ name }) {
        return name?.replace(/^@fixtures\/themed\/src\//, '') ?? '';
      },
    });
  });
});
