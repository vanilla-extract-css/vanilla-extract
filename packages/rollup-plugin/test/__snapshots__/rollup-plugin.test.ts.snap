// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin should build with preserveModules 1`] = `
Array [
  Array [
    "assets/src/shared.css.ts.vanilla-82a38ac6.css",
    ".shared__9k2war0 {
  box-shadow: 0 0 5px red;
}
body {
  background-color: skyblue;
}",
  ],
  Array [
    "assets/src/styles.css.ts.vanilla-3addedd5.css",
    "@font-face {
  src: local(\\"Impact\\");
  font-family: \\"styles__g7vce90\\";
}
@font-face {
  src: local(\\"Comic Sans MS\\");
  font-family: MyGlobalComicSans;
}
.styles__g7vce91 {
  display: flex;
  flex-direction: column;
  gap: var(--space-2__6cbmls4);
  padding: var(--space-3__6cbmls5);
}
.styles__g7vce92 {
  z-index: 1;
  position: relative;
}
.styles__g7vce93 {
  font-family: \\"styles__g7vce90\\";
  background-color: var(--colors-backgroundColor__6cbmls1, \\"THIS FALLBACK VALUE SHOULD NEVER BE USED\\");
  color: var(--colors-text__6cbmls2);
  border-radius: 9999px;
}
.themes__6cbmls6 .themes__6cbmls0 .styles__g7vce91 .styles__g7vce93 {
  font-family: MyGlobalComicSans;
  outline: 5px solid red;
}
body .styles__g7vce92:after {
  content: 'I am content';
}
html .styles_1\\\\/2__g7vce96 {
  opacity: var(--g7vce94, 0.5);
}
html .styles_1\\\\/4__g7vce97 {
  opacity: var(--g7vce94, var(--g7vce95, 0.25));
}
@media only screen and (min-width: 500px) {
  .styles__g7vce91 {
    border: 1px solid var(--colors-backgroundColor__6cbmls1);
  }
  .styles__g7vce93 {
    padding: var(--space-1__6cbmls3);
  }
}
@media only screen and (min-width: 1000px) {
  .styles__g7vce93 {
    padding: var(--space-2__6cbmls4);
  }
}",
  ],
  Array [
    "assets/src/themes.css.ts.vanilla-becd40aa.css",
    ":root, .themes__6cbmls0 {
  --colors-backgroundColor__6cbmls1: blue;
  --colors-text__6cbmls2: white;
  --space-1__6cbmls3: 4px;
  --space-2__6cbmls4: 8px;
  --space-3__6cbmls5: 12px;
}
.themes__6cbmls6 {
  --colors-backgroundColor__6cbmls1: green;
  --colors-text__6cbmls2: white;
  --space-1__6cbmls3: 8px;
  --space-2__6cbmls4: 12px;
  --space-3__6cbmls5: 16px;
}
.themes__6cbmls7 {
  --colors-backgroundColor__6cbmls1: pink;
  --colors-text__6cbmls2: purple;
  --space-1__6cbmls3: 6px;
  --space-2__6cbmls4: 12px;
  --space-3__6cbmls5: 18px;
}
@media screen and (min-width: 768px) {
  .themes__6cbmls7 {
    --colors-backgroundColor__6cbmls1: purple;
    --colors-text__6cbmls2: pink;
  }
}",
  ],
  Array [
    "src/index.js",
    "import { assignInlineVars, setElementVars } from '@vanilla-extract/dynamic';
import { vars, altTheme, theme, responsiveTheme } from './themes.css.js';
import { container, button, opacity } from './styles.css.js';
import { shadow } from './shared.css.js';
import testNodes from '../test-nodes.json.js';

const inlineTheme = assignInlineVars(vars, {
  colors: {
    backgroundColor: \\"orange\\",
    text: \\"black\\"
  },
  space: {
    1: \\"4px\\",
    2: \\"8px\\",
    3: \\"12px\\"
  }
});
function render() {
  document.body.innerHTML = \`
  <div id=\\"\${testNodes.root}\\" class=\\"\${shadow}\\"> 
    Root theme
    <div id=\\"\${testNodes.rootContainer}\\" class=\\"\${container}\\">
      <button id=\\"\${testNodes.rootButton}\\" class=\\"\${button}\\">Main theme button</button>
      <div class=\\"\${altTheme}\\"> 
        Alt theme
        <div id=\\"\${testNodes.altContainer}\\" class=\\"\${container}\\">
          <button id=\\"\${testNodes.altButton}\\" class=\\"\${button}\\">Alt theme button</button>
          <div class=\\"\${theme}\\"> 
            Back to root theme
            <div id=\\"\${testNodes.nestedRootContainer}\\" class=\\"\${container}\\">
              <button id=\\"\${testNodes.nestedRootButton}\\" class=\\"\${button}\\">Main theme button</button>
            <div style=\\"\${inlineTheme}\\">
              Inline theme
                <div id=\\"\${testNodes.inlineThemeContainer}\\" class=\\"\${container}\\">
                  <button id=\\"\${testNodes.inlineThemeButton}\\" class=\\"\${button} \${opacity[\\"1/2\\"]}\\">Inline theme button</button>
                  <div>
                  Dynamic vars
                    <div id=\\"\${testNodes.dynamicVarsContainer}\\" class=\\"\${container}\\">
                      <button id=\\"\${testNodes.dynamicVarsButton}\\" class=\\"\${button}\\">Dynamic vars button</button>
                  <div class=\\"\${responsiveTheme}\\">
              Responsive theme
                <div id=\\"\${testNodes.responsiveThemeContainer}\\" class=\\"\${container}\\">
                  <button id=\\"\${testNodes.responsiveThemeButton}\\" class=\\"\${button}\\">Responsive theme button</button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
\`;
  const dynamicVarsContainer = document.getElementById(testNodes.dynamicVarsContainer);
  if (!dynamicVarsContainer) {
    throw new Error(\\"Dynamic vars container not found.\\");
  }
  setElementVars(dynamicVarsContainer, vars, {
    colors: {
      backgroundColor: \\"transparent\\",
      text: \\"papayawhip\\"
    },
    space: {
      1: \\"5px\\",
      2: \\"10px\\",
      3: \\"15px\\"
    }
  });
  setElementVars(dynamicVarsContainer, {
    [vars.colors.backgroundColor]: \\"darksalmon\\"
  });
}
render();
",
  ],
  Array [
    "src/shared.css.js",
    "import './../assets/src/shared.css.ts.vanilla-82a38ac6.css';

var shadow = \\"shared__9k2war0\\";

export { shadow };
",
  ],
  Array [
    "src/styles.css.js",
    "import './../assets/src/shared.css.ts.vanilla-82a38ac6.css';
import './../assets/src/themes.css.ts.vanilla-becd40aa.css';
import './../assets/src/styles.css.ts.vanilla-3addedd5.css';

var button = \\"styles__g7vce93 shared__9k2war0 styles__g7vce92\\";
var container = \\"styles__g7vce91\\";
var opacity = {\\"1/2\\": \\"styles_1/2__g7vce96\\", \\"1/4\\": \\"styles_1/4__g7vce97\\"};

export { button, container, opacity };
",
  ],
  Array [
    "src/themes.css.js",
    "import './../assets/src/themes.css.ts.vanilla-becd40aa.css';

var altTheme = \\"themes__6cbmls6\\";
var responsiveTheme = \\"themes__6cbmls7\\";
var theme = \\"themes__6cbmls0\\";
var vars = {colors: {backgroundColor: \\"var(--colors-backgroundColor__6cbmls1)\\", text: \\"var(--colors-text__6cbmls2)\\"}, space: {\\"1\\": \\"var(--space-1__6cbmls3)\\", \\"2\\": \\"var(--space-2__6cbmls4)\\", \\"3\\": \\"var(--space-3__6cbmls5)\\"}};

export { altTheme, responsiveTheme, theme, vars };
",
  ],
  Array [
    "test-nodes.json.js",
    "var root = \\"root\\";
var rootContainer = \\"rootContainer\\";
var rootButton = \\"rootButton\\";
var altContainer = \\"altContainer\\";
var altButton = \\"altButton\\";
var nestedRootContainer = \\"nestedRootContainer\\";
var nestedRootButton = \\"nestedRootButton\\";
var inlineThemeContainer = \\"inlineThemeContainer\\";
var inlineThemeButton = \\"inlineThemeButton\\";
var dynamicVarsContainer = \\"dynamicVarsContainer\\";
var dynamicVarsButton = \\"dynamicVarsButton\\";
var responsiveThemeContainer = \\"responsiveThemeContainer\\";
var responsiveThemeButton = \\"responsiveThemeButton\\";
var testNodes = {
	root: root,
	rootContainer: rootContainer,
	rootButton: rootButton,
	altContainer: altContainer,
	altButton: altButton,
	nestedRootContainer: nestedRootContainer,
	nestedRootButton: nestedRootButton,
	inlineThemeContainer: inlineThemeContainer,
	inlineThemeButton: inlineThemeButton,
	dynamicVarsContainer: dynamicVarsContainer,
	dynamicVarsButton: dynamicVarsButton,
	responsiveThemeContainer: responsiveThemeContainer,
	responsiveThemeButton: responsiveThemeButton
};

export { altButton, altContainer, testNodes as default, dynamicVarsButton, dynamicVarsContainer, inlineThemeButton, inlineThemeContainer, nestedRootButton, nestedRootContainer, responsiveThemeButton, responsiveThemeContainer, root, rootButton, rootContainer };
",
  ],
]
`;

exports[`rollup-plugin should build with preserveModules and assetFileNames 1`] = `
Array [
  Array [
    "src/index.js",
    "import { assignInlineVars, setElementVars } from '@vanilla-extract/dynamic';
import { vars, altTheme, theme, responsiveTheme } from './themes.css.js';
import { container, button, opacity } from './styles.css.js';
import { shadow } from './shared.css.js';
import testNodes from '../test-nodes.json.js';

const inlineTheme = assignInlineVars(vars, {
  colors: {
    backgroundColor: \\"orange\\",
    text: \\"black\\"
  },
  space: {
    1: \\"4px\\",
    2: \\"8px\\",
    3: \\"12px\\"
  }
});
function render() {
  document.body.innerHTML = \`
  <div id=\\"\${testNodes.root}\\" class=\\"\${shadow}\\"> 
    Root theme
    <div id=\\"\${testNodes.rootContainer}\\" class=\\"\${container}\\">
      <button id=\\"\${testNodes.rootButton}\\" class=\\"\${button}\\">Main theme button</button>
      <div class=\\"\${altTheme}\\"> 
        Alt theme
        <div id=\\"\${testNodes.altContainer}\\" class=\\"\${container}\\">
          <button id=\\"\${testNodes.altButton}\\" class=\\"\${button}\\">Alt theme button</button>
          <div class=\\"\${theme}\\"> 
            Back to root theme
            <div id=\\"\${testNodes.nestedRootContainer}\\" class=\\"\${container}\\">
              <button id=\\"\${testNodes.nestedRootButton}\\" class=\\"\${button}\\">Main theme button</button>
            <div style=\\"\${inlineTheme}\\">
              Inline theme
                <div id=\\"\${testNodes.inlineThemeContainer}\\" class=\\"\${container}\\">
                  <button id=\\"\${testNodes.inlineThemeButton}\\" class=\\"\${button} \${opacity[\\"1/2\\"]}\\">Inline theme button</button>
                  <div>
                  Dynamic vars
                    <div id=\\"\${testNodes.dynamicVarsContainer}\\" class=\\"\${container}\\">
                      <button id=\\"\${testNodes.dynamicVarsButton}\\" class=\\"\${button}\\">Dynamic vars button</button>
                  <div class=\\"\${responsiveTheme}\\">
              Responsive theme
                <div id=\\"\${testNodes.responsiveThemeContainer}\\" class=\\"\${container}\\">
                  <button id=\\"\${testNodes.responsiveThemeButton}\\" class=\\"\${button}\\">Responsive theme button</button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
\`;
  const dynamicVarsContainer = document.getElementById(testNodes.dynamicVarsContainer);
  if (!dynamicVarsContainer) {
    throw new Error(\\"Dynamic vars container not found.\\");
  }
  setElementVars(dynamicVarsContainer, vars, {
    colors: {
      backgroundColor: \\"transparent\\",
      text: \\"papayawhip\\"
    },
    space: {
      1: \\"5px\\",
      2: \\"10px\\",
      3: \\"15px\\"
    }
  });
  setElementVars(dynamicVarsContainer, {
    [vars.colors.backgroundColor]: \\"darksalmon\\"
  });
}
render();
",
  ],
  Array [
    "src/shared.css.js",
    "import './shared.css.ts.vanilla.css';

var shadow = \\"shared__9k2war0\\";

export { shadow };
",
  ],
  Array [
    "src/shared.css.ts.vanilla.css",
    ".shared__9k2war0 {
  box-shadow: 0 0 5px red;
}
body {
  background-color: skyblue;
}",
  ],
  Array [
    "src/styles.css.js",
    "import './shared.css.ts.vanilla.css';
import './themes.css.ts.vanilla.css';
import './styles.css.ts.vanilla.css';

var button = \\"styles__g7vce93 shared__9k2war0 styles__g7vce92\\";
var container = \\"styles__g7vce91\\";
var opacity = {\\"1/2\\": \\"styles_1/2__g7vce96\\", \\"1/4\\": \\"styles_1/4__g7vce97\\"};

export { button, container, opacity };
",
  ],
  Array [
    "src/styles.css.ts.vanilla.css",
    "@font-face {
  src: local(\\"Impact\\");
  font-family: \\"styles__g7vce90\\";
}
@font-face {
  src: local(\\"Comic Sans MS\\");
  font-family: MyGlobalComicSans;
}
.styles__g7vce91 {
  display: flex;
  flex-direction: column;
  gap: var(--space-2__6cbmls4);
  padding: var(--space-3__6cbmls5);
}
.styles__g7vce92 {
  z-index: 1;
  position: relative;
}
.styles__g7vce93 {
  font-family: \\"styles__g7vce90\\";
  background-color: var(--colors-backgroundColor__6cbmls1, \\"THIS FALLBACK VALUE SHOULD NEVER BE USED\\");
  color: var(--colors-text__6cbmls2);
  border-radius: 9999px;
}
.themes__6cbmls6 .themes__6cbmls0 .styles__g7vce91 .styles__g7vce93 {
  font-family: MyGlobalComicSans;
  outline: 5px solid red;
}
body .styles__g7vce92:after {
  content: 'I am content';
}
html .styles_1\\\\/2__g7vce96 {
  opacity: var(--g7vce94, 0.5);
}
html .styles_1\\\\/4__g7vce97 {
  opacity: var(--g7vce94, var(--g7vce95, 0.25));
}
@media only screen and (min-width: 500px) {
  .styles__g7vce91 {
    border: 1px solid var(--colors-backgroundColor__6cbmls1);
  }
  .styles__g7vce93 {
    padding: var(--space-1__6cbmls3);
  }
}
@media only screen and (min-width: 1000px) {
  .styles__g7vce93 {
    padding: var(--space-2__6cbmls4);
  }
}",
  ],
  Array [
    "src/themes.css.js",
    "import './themes.css.ts.vanilla.css';

var altTheme = \\"themes__6cbmls6\\";
var responsiveTheme = \\"themes__6cbmls7\\";
var theme = \\"themes__6cbmls0\\";
var vars = {colors: {backgroundColor: \\"var(--colors-backgroundColor__6cbmls1)\\", text: \\"var(--colors-text__6cbmls2)\\"}, space: {\\"1\\": \\"var(--space-1__6cbmls3)\\", \\"2\\": \\"var(--space-2__6cbmls4)\\", \\"3\\": \\"var(--space-3__6cbmls5)\\"}};

export { altTheme, responsiveTheme, theme, vars };
",
  ],
  Array [
    "src/themes.css.ts.vanilla.css",
    ":root, .themes__6cbmls0 {
  --colors-backgroundColor__6cbmls1: blue;
  --colors-text__6cbmls2: white;
  --space-1__6cbmls3: 4px;
  --space-2__6cbmls4: 8px;
  --space-3__6cbmls5: 12px;
}
.themes__6cbmls6 {
  --colors-backgroundColor__6cbmls1: green;
  --colors-text__6cbmls2: white;
  --space-1__6cbmls3: 8px;
  --space-2__6cbmls4: 12px;
  --space-3__6cbmls5: 16px;
}
.themes__6cbmls7 {
  --colors-backgroundColor__6cbmls1: pink;
  --colors-text__6cbmls2: purple;
  --space-1__6cbmls3: 6px;
  --space-2__6cbmls4: 12px;
  --space-3__6cbmls5: 18px;
}
@media screen and (min-width: 768px) {
  .themes__6cbmls7 {
    --colors-backgroundColor__6cbmls1: purple;
    --colors-text__6cbmls2: pink;
  }
}",
  ],
  Array [
    "test-nodes.json.js",
    "var root = \\"root\\";
var rootContainer = \\"rootContainer\\";
var rootButton = \\"rootButton\\";
var altContainer = \\"altContainer\\";
var altButton = \\"altButton\\";
var nestedRootContainer = \\"nestedRootContainer\\";
var nestedRootButton = \\"nestedRootButton\\";
var inlineThemeContainer = \\"inlineThemeContainer\\";
var inlineThemeButton = \\"inlineThemeButton\\";
var dynamicVarsContainer = \\"dynamicVarsContainer\\";
var dynamicVarsButton = \\"dynamicVarsButton\\";
var responsiveThemeContainer = \\"responsiveThemeContainer\\";
var responsiveThemeButton = \\"responsiveThemeButton\\";
var testNodes = {
	root: root,
	rootContainer: rootContainer,
	rootButton: rootButton,
	altContainer: altContainer,
	altButton: altButton,
	nestedRootContainer: nestedRootContainer,
	nestedRootButton: nestedRootButton,
	inlineThemeContainer: inlineThemeContainer,
	inlineThemeButton: inlineThemeButton,
	dynamicVarsContainer: dynamicVarsContainer,
	dynamicVarsButton: dynamicVarsButton,
	responsiveThemeContainer: responsiveThemeContainer,
	responsiveThemeButton: responsiveThemeButton
};

export { altButton, altContainer, testNodes as default, dynamicVarsButton, dynamicVarsContainer, inlineThemeButton, inlineThemeContainer, nestedRootButton, nestedRootContainer, responsiveThemeButton, responsiveThemeContainer, root, rootButton, rootContainer };
",
  ],
]
`;

exports[`rollup-plugin should build without preserveModules 1`] = `
Array [
  Array [
    "assets/src/shared.css.ts.vanilla-82a38ac6.css",
    ".shared__9k2war0 {
  box-shadow: 0 0 5px red;
}
body {
  background-color: skyblue;
}",
  ],
  Array [
    "assets/src/styles.css.ts.vanilla-3addedd5.css",
    "@font-face {
  src: local(\\"Impact\\");
  font-family: \\"styles__g7vce90\\";
}
@font-face {
  src: local(\\"Comic Sans MS\\");
  font-family: MyGlobalComicSans;
}
.styles__g7vce91 {
  display: flex;
  flex-direction: column;
  gap: var(--space-2__6cbmls4);
  padding: var(--space-3__6cbmls5);
}
.styles__g7vce92 {
  z-index: 1;
  position: relative;
}
.styles__g7vce93 {
  font-family: \\"styles__g7vce90\\";
  background-color: var(--colors-backgroundColor__6cbmls1, \\"THIS FALLBACK VALUE SHOULD NEVER BE USED\\");
  color: var(--colors-text__6cbmls2);
  border-radius: 9999px;
}
.themes__6cbmls6 .themes__6cbmls0 .styles__g7vce91 .styles__g7vce93 {
  font-family: MyGlobalComicSans;
  outline: 5px solid red;
}
body .styles__g7vce92:after {
  content: 'I am content';
}
html .styles_1\\\\/2__g7vce96 {
  opacity: var(--g7vce94, 0.5);
}
html .styles_1\\\\/4__g7vce97 {
  opacity: var(--g7vce94, var(--g7vce95, 0.25));
}
@media only screen and (min-width: 500px) {
  .styles__g7vce91 {
    border: 1px solid var(--colors-backgroundColor__6cbmls1);
  }
  .styles__g7vce93 {
    padding: var(--space-1__6cbmls3);
  }
}
@media only screen and (min-width: 1000px) {
  .styles__g7vce93 {
    padding: var(--space-2__6cbmls4);
  }
}",
  ],
  Array [
    "assets/src/themes.css.ts.vanilla-becd40aa.css",
    ":root, .themes__6cbmls0 {
  --colors-backgroundColor__6cbmls1: blue;
  --colors-text__6cbmls2: white;
  --space-1__6cbmls3: 4px;
  --space-2__6cbmls4: 8px;
  --space-3__6cbmls5: 12px;
}
.themes__6cbmls6 {
  --colors-backgroundColor__6cbmls1: green;
  --colors-text__6cbmls2: white;
  --space-1__6cbmls3: 8px;
  --space-2__6cbmls4: 12px;
  --space-3__6cbmls5: 16px;
}
.themes__6cbmls7 {
  --colors-backgroundColor__6cbmls1: pink;
  --colors-text__6cbmls2: purple;
  --space-1__6cbmls3: 6px;
  --space-2__6cbmls4: 12px;
  --space-3__6cbmls5: 18px;
}
@media screen and (min-width: 768px) {
  .themes__6cbmls7 {
    --colors-backgroundColor__6cbmls1: purple;
    --colors-text__6cbmls2: pink;
  }
}",
  ],
  Array [
    "index.js",
    "import { assignInlineVars, setElementVars } from '@vanilla-extract/dynamic';
import './assets/src/themes.css.ts.vanilla-becd40aa.css';
import './assets/src/shared.css.ts.vanilla-82a38ac6.css';
import './assets/src/styles.css.ts.vanilla-3addedd5.css';

var altTheme = \\"themes__6cbmls6\\";
var responsiveTheme = \\"themes__6cbmls7\\";
var theme = \\"themes__6cbmls0\\";
var vars = {colors: {backgroundColor: \\"var(--colors-backgroundColor__6cbmls1)\\", text: \\"var(--colors-text__6cbmls2)\\"}, space: {\\"1\\": \\"var(--space-1__6cbmls3)\\", \\"2\\": \\"var(--space-2__6cbmls4)\\", \\"3\\": \\"var(--space-3__6cbmls5)\\"}};

var button = \\"styles__g7vce93 shared__9k2war0 styles__g7vce92\\";
var container = \\"styles__g7vce91\\";
var opacity = {\\"1/2\\": \\"styles_1/2__g7vce96\\", \\"1/4\\": \\"styles_1/4__g7vce97\\"};

var shadow = \\"shared__9k2war0\\";

var root = \\"root\\";
var rootContainer = \\"rootContainer\\";
var rootButton = \\"rootButton\\";
var altContainer = \\"altContainer\\";
var altButton = \\"altButton\\";
var nestedRootContainer = \\"nestedRootContainer\\";
var nestedRootButton = \\"nestedRootButton\\";
var inlineThemeContainer = \\"inlineThemeContainer\\";
var inlineThemeButton = \\"inlineThemeButton\\";
var dynamicVarsContainer = \\"dynamicVarsContainer\\";
var dynamicVarsButton = \\"dynamicVarsButton\\";
var responsiveThemeContainer = \\"responsiveThemeContainer\\";
var responsiveThemeButton = \\"responsiveThemeButton\\";
var testNodes = {
	root: root,
	rootContainer: rootContainer,
	rootButton: rootButton,
	altContainer: altContainer,
	altButton: altButton,
	nestedRootContainer: nestedRootContainer,
	nestedRootButton: nestedRootButton,
	inlineThemeContainer: inlineThemeContainer,
	inlineThemeButton: inlineThemeButton,
	dynamicVarsContainer: dynamicVarsContainer,
	dynamicVarsButton: dynamicVarsButton,
	responsiveThemeContainer: responsiveThemeContainer,
	responsiveThemeButton: responsiveThemeButton
};

const inlineTheme = assignInlineVars(vars, {
  colors: {
    backgroundColor: \\"orange\\",
    text: \\"black\\"
  },
  space: {
    1: \\"4px\\",
    2: \\"8px\\",
    3: \\"12px\\"
  }
});
function render() {
  document.body.innerHTML = \`
  <div id=\\"\${testNodes.root}\\" class=\\"\${shadow}\\"> 
    Root theme
    <div id=\\"\${testNodes.rootContainer}\\" class=\\"\${container}\\">
      <button id=\\"\${testNodes.rootButton}\\" class=\\"\${button}\\">Main theme button</button>
      <div class=\\"\${altTheme}\\"> 
        Alt theme
        <div id=\\"\${testNodes.altContainer}\\" class=\\"\${container}\\">
          <button id=\\"\${testNodes.altButton}\\" class=\\"\${button}\\">Alt theme button</button>
          <div class=\\"\${theme}\\"> 
            Back to root theme
            <div id=\\"\${testNodes.nestedRootContainer}\\" class=\\"\${container}\\">
              <button id=\\"\${testNodes.nestedRootButton}\\" class=\\"\${button}\\">Main theme button</button>
            <div style=\\"\${inlineTheme}\\">
              Inline theme
                <div id=\\"\${testNodes.inlineThemeContainer}\\" class=\\"\${container}\\">
                  <button id=\\"\${testNodes.inlineThemeButton}\\" class=\\"\${button} \${opacity[\\"1/2\\"]}\\">Inline theme button</button>
                  <div>
                  Dynamic vars
                    <div id=\\"\${testNodes.dynamicVarsContainer}\\" class=\\"\${container}\\">
                      <button id=\\"\${testNodes.dynamicVarsButton}\\" class=\\"\${button}\\">Dynamic vars button</button>
                  <div class=\\"\${responsiveTheme}\\">
              Responsive theme
                <div id=\\"\${testNodes.responsiveThemeContainer}\\" class=\\"\${container}\\">
                  <button id=\\"\${testNodes.responsiveThemeButton}\\" class=\\"\${button}\\">Responsive theme button</button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
\`;
  const dynamicVarsContainer = document.getElementById(testNodes.dynamicVarsContainer);
  if (!dynamicVarsContainer) {
    throw new Error(\\"Dynamic vars container not found.\\");
  }
  setElementVars(dynamicVarsContainer, vars, {
    colors: {
      backgroundColor: \\"transparent\\",
      text: \\"papayawhip\\"
    },
    space: {
      1: \\"5px\\",
      2: \\"10px\\",
      3: \\"15px\\"
    }
  });
  setElementVars(dynamicVarsContainer, {
    [vars.colors.backgroundColor]: \\"darksalmon\\"
  });
}
render();
",
  ],
]
`;
